src/main/java/org/hydev/mcpm/client/database/mirrors/MirrorSelectBoundary.java

```diff
@@ -63,4 +65,26 @@ default List<Pair<Mirror, Integer>> pingMirrors() throws IOException
      * @param mirror A selected mirror
      */
     void setSelectedMirror(Mirror mirror) throws IOException;
+
+    /**
+     * Get the selected mirror as a URI supplier
+     *
+     * @return URI supplier function
+     */
+    default Supplier<URI> selectedMirrorSupplier()
```

Comment:
Also, I think this is a bit of an abuse of default methods. IMO interfaces should be behaviour-less, small enough to get a good sense of what you need to implement. You're also forgoing static dispatch.

I would recommend moving some of these out into a static class.